src/singletons/Toasts.cpp

```diff
@@ -64,152 +42,108 @@ QString Toasts::findStringFromReaction(
+                                      const QString &channelName)
 {
-private:
-    QString channelName_;
-    Platform platform_;
-
-public:
-    CustomHandler(QString channelName, Platform p)
-        : channelName_(channelName)
-        , platform_(p)
+    QString bottomText = "";
+    if (static_cast<ToastReaction>(getSettings()->openFromToast.getValue()) !=
+        ToastReaction::DontOpen)
     {
+        QString mode =
+            Toasts::findStringFromReaction(getSettings()->openFromToast)
+                .toLower();
+
+        bottomText = "Click here to " + mode;
     }
-    void toastActivated() const
-    {
-        QString link;
-        auto toastReaction =
-            static_cast<ToastReaction>(getSettings()->openFromToast.getValue());
 
-        switch (toastReaction)
+    auto callback = [channelName]() {
+        switch (
+            static_cast<ToastReaction>(getSettings()->openFromToast.getValue()))
         {
             case ToastReaction::OpenInBrowser:
-                if (platform_ == Platform::Twitch)
-                {
-                    link = "http://www.twitch.tv/" + channelName_;
-                }
-                QDesktopServices::openUrl(QUrl(link));
+                QDesktopServices::openUrl(
+                    QUrl("https://twitch.tv/" + channelName));
                 break;
             case ToastReaction::OpenInPlayer:
-                if (platform_ == Platform::Twitch)
-                {
-                    link =
-                        "https://player.twitch.tv/?parent=twitch.tv&channel=" +
-                        channelName_;
-                }
-                QDesktopServices::openUrl(QUrl(link));
+                QDesktopServices::openUrl(
+                    QUrl("https://player.twitch.tv/?parent=twitch.tv&channel=" +
+                         channelName));
                 break;
-            case ToastReaction::OpenInStreamlink: {
-                openStreamlinkForChannel(channelName_);
+            case ToastReaction::OpenInStreamlink:
+                openStreamlinkForChannel(channelName);
                 break;
-            }
-                // the fourth and last option is "don't open"
-                // in this case obviously nothing should happen
         }
-    }
-
-    void toastActivated(int actionIndex) const
-    {
-    }
-
-    void toastFailed() const
-    {
-    }
+    };
 
-    void toastDismissed(WinToastDismissalReason state) const
-    {
-    }
-};
+    NetworkRequest::twitchRequest(url)
+        .onSuccess(
+            [channelName, bottomText, callback, this](auto result) -> Outcome {
+                const auto data = result.getData();
+
+                QPixmap avatar;
+                avatar.loadFromData(data);
+
+                getApp()->notifications->addNotification(
+                    makeLayout(
+                        avatar,
+                        QString("<b>" + channelName + "</b> just went live!"),
```

Comment:
This just looks cleaner
```suggestion
                        QString("<b>%1</b> just went live!").arg(channelName),
```